lib/rabbit_feed/event_definitions.rb

```diff
@@ -51,19 +51,31 @@ def defined_as &block
         @definition = block.call if block.present?
       end
 
-      def payload
-        ([
+      def payload_schema
+        { name: "#{name}_payload", type: 'record', fields: fields.map(&:schema) }
+      end
+
+      def metadata_schema
+        { name: 'event_metadata', type: 'record', fields: [
           (Field.new 'application',    'string', 'The name of the application that created the event'),
           (Field.new 'host',           'string', 'The hostname of the server on which the event was created'),
           (Field.new 'environment',    'string', 'The environment in which the event was created'),
-          (Field.new 'version',        'string', 'The version of the event'),
+          (Field.new 'version',        'string', 'The version of the event payload'),
+          (Field.new 'gem_version',    'string', 'The version of rabbit_feed by which the event was created'),
```

Comment:
I see your point. My thinking is that if the event structure changes, that would require a change to the gem version number (I should document this guideline, though). Of course, other types of changes would also require a change to the gem version number (e.g. change to the API). You're proposing to version the event structure explicitly. The one thing I like about using the gem version number is that it tells you in a bit more granularity the version of rabbit feed that was used to generate the event. I'm inclined to keep this as is (with the addition of documenting this process), unless you feel strongly.